clang/lib/3C/ConstraintVariables.cpp

```diff
@@ -1047,6 +1060,7 @@ FunctionVariableConstraint::FunctionVariableConstraint(const Type *Ty,
     auto PSL = PersistentSourceLoc::mkPSL(D, *TmpCtx);
     FileName = PSL.getFileName();
     IsFunctionPtr = false;
+    TypeParams = FD->getNumTypeVars();
```

Comment:
What is `FD`? Is this the case where the generic is already in the code you are analyzing?

Reply:
That's correct. The code currently uses this data as if it were a bool, generic or not. I didn't know of this method last time and set it in a more complex way below.